src/sources/soundsourceproxy.cpp

```diff
@@ -464,12 +464,11 @@ void SoundSourceProxy::loadTrackMetadataAndCoverArt(
         parsedFromFile = true;
         if (!coverImg.isNull()) {
             // Cover image has been parsed from the file
-            coverArt.image = coverImg;
             // TODO() here we may introduce a duplicate hash code
-            coverArt.info.hash = CoverArtUtils::calculateHash(coverArt.image);
-            coverArt.info.coverLocation = QString();
-            coverArt.info.type = CoverInfo::METADATA;
-            coverArt.info.source = CoverInfo::GUESSED;
+            coverInfoRelative.hash = CoverArtUtils::calculateHash(coverImg);
```

Comment:
Why not encapsulate this code as a member functions in the CoverInfo classes? Setting all members individually violates the encapsulation principle. We should follow OO design principles if we decide to use a class hierarchy for CoverInfo ;) Maybe adding those member functions provides new insights if our OO design is appropriate and useful.